src/Workspaces/Core/Portable/Diagnostics/Extensions.cs

```diff
@@ -57,6 +58,15 @@ public static DiagnosticData ToDiagnosticData(this Diagnostic diagnostic, Projec
                 return DiagnosticData.Create(project.GetDocument(diagnostic.Location.SourceTree), diagnostic);
             }
 
+            if (diagnostic.Location.Kind == LocationKind.ExternalFile)
```

Comment:
This method is only called by the SuggestActionProviders (through `codeFix.GetPrimaryDiagnosticData()`) and so TypeScript shouldn't be affected.